security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -0,0 +1,195 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package cache is the in-memory secret store.
+package cache
+
+import (
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"istio.io/istio/pkg/log"
+	"istio.io/istio/security/pkg/pki/util"
+)
+
+// The size of a private key for a leaf certificate.
+const keySize = 2048
+
+// CAClient interface defines the clients need to implement to talk to CA for CSR.
+// TODO(quanlin): CAClient here is a placeholder, will move it to separated pkg.
+type CAClient interface {
```

Comment:
Here we actually assemble and sending CSR, not performing signing logic. Can we rename it to something more accurate? Like `FetchKeyCert` or something else?

https://github.com/istio/istio/blob/master/security/pkg/pki/ca/ca.go#L59 want to distinguish with `Sign` here.

Reply:
cert is generated locally, so fetch** sounds a bit confusing to me. The caClient implementation need to assemble CSR then send request to corresponding CA, I feel at abstract level 'sign' is fine(interface hides implementation details of signing locally or remotely)

Reply:
To be precise, cert is generated locally but it's the Google CA signed that.

Just looking it here, I agree with your point that `Sign` can treated as an abstract operation. But if we look at other places,
- `pkg/caclient` sends request 
- In `pkg/pki/ca/ca.go`, the Sign means just a locally Sign, no CSR involved.
- Here the `Sign` means both.

Even we develop separate libraries, I hope we can be consistent of the terminology otherwise it'll very hard to tack & understand the code.

@myidpt

Reply:
renamed to CSRSign.